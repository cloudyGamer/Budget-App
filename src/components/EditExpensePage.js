 
import React from 'react'; 
import { connect } from 'react-redux';
import { startEditExpense,editExpense,removeExpense, startRemoveExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

export class EditExpensePage extends React.Component {
     onSubmit = (expense) => {
          //console.log("updated",expense);
          this.props.startEditExpense(this.props.expense.id,expense);
          this.props.history.push(`/`);
     };
     
     onRemove = () => {
          this.props.startRemoveExpense({id: this.props.expense.id});
          this.props.history.push(`/`);
     };
     render(){
          return(
               <div>
                 <h1>Expense Form</h1>    
                    <ExpenseForm xmlns="http://www.w3.org/1999/xhtml"
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                    /> 
                    <button 
                    onClick={this.onRemove}>Remove</button>
               </div>  
            );
     }
};
const mapDispatchToProps = (dispatch,props) => ({
     startEditExpense: (expenseid,expense) => dispatch(startEditExpense(expenseid,expense)),
     startRemoveExpense: (expense) => dispatch(startRemoveExpense(expense))
});

const mapStateToProps = (state,props) => {
     return {
          expense : state.expenses.find((expense) => expense.id===props.match.params.id)
     };
};

 export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage);