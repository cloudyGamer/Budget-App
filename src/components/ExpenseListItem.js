
import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
 //description //createdAt //amount
 //use map - the map goes in expense list. pass the individual items as props
// load a locale
numeral.register('locale', 'irl', {
    delimiters: {
        thousands: ' ',
        decimal: ','
    },
    abbreviations: {
        thousand: 'k',
        million: 'm',
        billion: 'b',
        trillion: 't'
    },
    ordinal : function (number) {
        return number === 1 ? 'st' : 'th';
    },
    currency: {
        symbol: 'â‚¬'
    }
});

// switch between locales
numeral.locale('irl');
export const ExpenseListItem = ({dispatch, description, amount, createdAt, id}) => (
  <div xmlns="http://www.w3.org/1999/xhtml">
  <Link to={`/edit/${id}`}><h3>{description}</h3></Link>
    <p>
    {numeral(amount/100).format('$0,0.00')} 
    - 
    {moment(createdAt).format('MMMM Do, YYYY')}</p>

  </div>  

  );

export default ExpenseListItem;
