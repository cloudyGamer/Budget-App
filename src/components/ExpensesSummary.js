import React from 'react';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import { connect } from 'react-redux';

//export const ExpensesSummary = (props) => (
//   {/*  <div>
//   {
//   props.expenses.length !== 0 &&
//     <div>
//   <a>Displaying</a>  <a>{props.expenses.length}</a> { props.expenses.length === 1 ? (<a>expense</a> ) : (<a>expenses</a>)}
//      <a> with a value of </a> 
//       <a>{props.expenses
//       .map((expense)=>expense.amount)
//       .reduce((total,num) => total+num, 0)}
//       </a>
//     </div> 
//          
//          }
//   </div>*/}  
//
//   );

     
export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
     const expenseWord = expenseCount ===1 ? 'expense':'expenses';
     const formattedTotal = numeral(expensesTotal/100).format('â‚¬0,0.00');
     return (
       <div>
       <h1>Viewing {expenseCount} {expenseWord} totalling {formattedTotal} euro</h1>
       </div>  
       );
};  

const mapStateToProps = (state) => {
     const visibleExpenses = selectExpenses(state.expenses, state.filters);
      return {
           expenseCount: visibleExpenses.length,
           expensesTotal: selectExpensesTotal(visibleExpenses)
           
      };
 };

export default connect(mapStateToProps)(ExpensesSummary);
