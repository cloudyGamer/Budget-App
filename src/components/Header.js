 /* 
  * To change this license header, choose License Headers in Project Properties.
  * To change this template file, choose Tools | Templates
  * and open the template in the editor.
  */
 
import React from 'react'; 
 import {Link,NavLink} from 'react-router-dom';
const Header = () => (
  <header>
  <h1>Expensify App</h1>
  <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
  <NavLink to="/create" activeClassName="is-active">Add Expense</NavLink>
  <NavLink to="/help" activeClassName="is-active">Help</NavLink>
  </header>
  );

 export default Header;