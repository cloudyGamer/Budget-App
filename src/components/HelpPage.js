 /* 
  * To change this license header, choose License Headers in Project Properties.
  * To change this template file, choose Tools | Templates
  * and open the template in the editor.
  */
  
import React from 'react'; 
 const HelpPage = () => (
  <div>
  This is from my HelpPage  component
  </div>  

  );
 
 export default HelpPage;