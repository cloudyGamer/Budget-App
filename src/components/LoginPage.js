import React from 'react'; 
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

/*export class LoginPage extends React.Component {
          constructor(props) {   
               super(props);
               
               this.state = {
               email: '',
               password: '',
               error: ''
               }
          }
        
        
          onEmailChange = (e) => {
              const email = e.target.value;
              this.setState(() => ({ email }));
          };
          
          onPasswordChange = (e) => {
               const password = e.target.value;
               this.setState(() => ({ password }));
          };
          
          onSubmit = (e) => {
               e.preventDefault();
               if (!this.state.email||!this.state.password) {
                    this.setState(() => ({ error: "please provide Email and Password" }));
                    //set error state equal to "please provide email and amount"
               }else {
                    this.setState(() => ({ error: '' }));
                    //you no longer have a props.submit here. import it
                    const details = {
                         email:this.state.email,
                         password: this.state.password
                    };
                    this.props.startLogin(); //login fnc
                    this.props.history.push(`/`);
               }
          }
          
        
        render() {
        
        return(<div  xmlns="http://www.w3.org/1999/xhtml">
                    {this.state.error&&<p>{this.state.error}</p>}
                  <form onSubmit={this.onSubmit}>
                    <input 
                         type="text"
                         placeholder="email"
                         autoFocus
                         value={this.state.email}
                         onChange={this.onEmailChange}
                    />
                   
                    <textarea
                         placeholder='Add a note for your expense (optional)'
                         value={this.state.password}
                         onChange={this.onPasswordChange}
                    />
                       <button>Login</button>
                  </form>  
               </div>);
    }  
    
}*/
export const LoginPage = ( { startLogin } ) => (
  <div  xmlns="http://www.w3.org/1999/xhtml">
  <button onClick={startLogin} >Log In</button>
  </div>
  );
const mapDispatchToProps = (dispatch) => ({
     startLogin: () => dispatch(startLogin())
});

//const mapStateToProps = (state,props) => {
//     return {
//          expense : state.expenses.find((expense) => expense.id===props.match.params.id)
//     };
//};

 export default connect(undefined,mapDispatchToProps)(LoginPage);