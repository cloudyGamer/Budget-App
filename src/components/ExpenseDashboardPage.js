 /* 
  * To change this license header, choose License Headers in Project Properties.
  * To change this template file, choose Tools | Templates
  * and open the template in the editor.
  */

import React from 'react'; 
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';
export const ExpenseDashboardPage = () => (
  <div xmlns="http://www.w3.org/1999/xhtml">
     <ExpensesSummary />
     <ExpenseListFilters />
     <ExpenseList />
  </div>  

  );

 export default ExpenseDashboardPage;