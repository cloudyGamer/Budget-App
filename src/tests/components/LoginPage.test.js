import React from 'react';
import { shallow } from 'enzyme'; 
import  { LoginPage }   from '../../components/LoginPage';
let wrapper;
beforeEach(() => {
      wrapper= shallow(<LoginPage
                         xmlns="http://www.w3.org/1999/xhtml" />);
});

test('should render login page',() => {
     expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
      const startLogin = jest.fn();
      const wrapper = shallow(<LoginPage startLogin={startLogin} xmlns="http://www.w3.org/1999/xhtml"/>);
      wrapper.find('button').simulate('click');
      expect(startLogin).toHaveBeenCalled();
 });