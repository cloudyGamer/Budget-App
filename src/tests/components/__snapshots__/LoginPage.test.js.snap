// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page 1`] = `
<div
  xmlns="http://www.w3.org/1999/xhtml"
>
  <button>
    Log In
  </button>
</div>
`;
